### ISSUE TITLE
[Debian OS] hooks/cryptroot patch undone by system update

### ISSUE THREAD

**Issue by tvdstaaij** [ViRb3/pi-encrypted-boot-ssh#18]
During a system update sometime in October which involved a new kernel package, my initramfs was regenerated and after a reboot the rootfs could no longer be unlocked through SSH. After inspection in a chroot it turned out that `/usr/share/initramfs-tools/hooks/cryptroot` had been overwritten and the patch was no longer there. I managed to fix it by re-applying the patch and regenerating the initramfs.

I should note that I am using a modified version of this guide with Debian, but given that Ubuntu and Raspberry Pi OS are derived from Debian I'd say that this could probably happen with those distributions as well.

When I find some time I will probably look into how exactly this happens and how it could be prevented. However if anyone else has experienced something like this before and/or any potential solutions come to mind, I would much appreciate your insights.

**Comment by ViRb3** [comment]
Since this is an embedded system, distros can vastly differ in how they handle things like boot and initramfs. Raspberry Pi OS for instance completely overhauled the boot process and initramfs handling in their Debian 11 to 12 update. I just released an update to this guide which was designed against the latest Raspberry Pi OS, the initramfs should integrate very nicely there. Apart from that, not much I can do to help I'm afraid. Good luck!

**Comment by tvdstaaij** [comment]
I looked into it, and it seems that although the patch had indeed been undone as I described, that wasn't actually the cause of my unbootable system. As you might expect, the patch is only really relevant during the initial setup. No idea what did cause the setup to break, but something must have gone awry during the automatic update and it just needed to be regenerated. If it happens again I'll inspect the bad initramfs to look for clues. Could very well be completely unrelated to cryptsetup.

I suspect the problem I described could still happen (on any OS) for the optional timeout patch to `/usr/share/cryptsetup/initramfs/bin/cryptroot-unlock`. Since this is not considered a config file, dpkg will be happy to overwrite it with a newer version from the respective package. Might be worth mentioning in the guide if you agree that this could be a problem.
